backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/KonanIr.proto

```diff
@@ -1,79 +1,145 @@
-        VARIABLE = 2;
-        CLASS = 3;
-        VALUE_PARAMETER = 4;
-        CONSTRUCTOR = 5;
-        TYPE_PARAMETER = 6;
-        RECEIVER = 7;
-        ACCESSOR = 8;
-        PROPERTY = 9;
-    }
-    // TODO: Use the string table index.
-    optional string name = 1;
-    required Kind kind = 2;
-    required UniqId uniq_id = 3;
-    required UniqId original_uniq_id = 4;
-    // index in fq name table
-    optional int32 class_or_package = 5;
-
-    // Descriptor kind specific fields.
-    // TODO: consider introducing specific messages
-    // for functions, variables etc.
-    optional KotlinType type = 6;
-    repeated KotlinDescriptor value_parameter = 7;
-    repeated KotlinDescriptor type_parameter = 8;
-    optional UniqId dispatch_receiver_uniq_id = 9;
-    optional UniqId extension_receiver_uniq_id = 10;
-    optional KotlinType extension_receiver_type = 11;
-    optional LocalDeclaration ir_local_declaration = 12;
+message DescriptorReference {
+    required string package_fq_name = 1;
+    required string class_fq_name = 2;
+    required string name = 3;
+    optional UniqId uniq_id = 4;
+    optional bool is_getter = 5 [default = false];
+    optional bool is_setter = 6 [default = false];
+    optional bool is_backing_field = 7 [default = false];
+    optional bool is_fake_override = 8 [default = false];
+    optional bool is_default_constructor = 9 [default = false];
+    optional bool is_enum_entry = 10 [default = false];
+    optional bool is_enum_special = 11 [default = false];
 }
 
 message UniqId {
-    required int64 index = 1;
+    required uint64 index = 1;
+    required bool isLocal = 2;
 }
 
-// Inner descriptors used for inlining.
-message DeclarationDescriptor {
-    // These are ProtoBuf.Function, ProtoBuf.Class etc
-    oneof descriptor {
-        Function function = 1;
-        Class clazz = 2;
-        Property property = 3;
-        Constructor constructor = 4;
-    }
+message Coordinates {
+    required int32 start_offset = 1;
+    required int32 end_offset = 2;
+}
+
+/* ------ Top Level---------------------------------------------- */
+
+message IrDeclarationContainer {
+    repeated IrDeclaration declaration = 1;
 }
-message LocalDeclaration {
-    required DeclarationDescriptor descriptor = 1;
+
+message FileEntry { // TODO: extend me.
+    required string name = 1;
+}
+
+message IrFile {
+    repeated UniqId declaration_id = 1;
+    required FileEntry file_entry = 2;
+    // TODO: we need a better string management. See metadata serialization as an example.
+    required string fq_name = 3;
+}
+
+message IrModule {
+    required string name = 1;
+    repeated IrFile file = 2;
+    required IrSymbolTable symbol_table = 3;
+}
+
+/* ------ IrSymbols --------------------------------------------- */
+
+enum IrSymbolKind {
+    FUNCTION_SYMBOL = 1;
+    CONSTRUCTOR_SYMBOL = 2;
+    ENUM_ENTRY_SYMBOL = 3;
+    FIELD_SYMBOL = 4;
+    VALUE_PARAMETER_SYMBOL = 5;
+    RETURN_TARGET_SYMBOL = 6;
```

Comment:
Not a real symbol kind actually. Maybe you mean `RETURNABLE_BLOCK_SYMBOL`? It doesn't exist before function inlining yet though.